### ISSUE TITLE
Swap out or remove Deprecated packages

### ISSUE THREAD

**Issue by Cruikshanks** [DEFRA/water-abstraction-team#22]
We are finding as we go through the code base that we have a number of deprecated packages.

They are no longer maintained so we need to get them out. If we don't, we expose ourselves to unfixed errors, vulnerabilities, and issues when we come to upgrade other packages.

As we find them we'll log them here and then note in the comments when we have tackled them.

- [ ] [Lodash](https://www.npmjs.com/package/lodash) - It's not actually deprecated but the community now advises against its use [You don't need Lodash](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore). We should be able to use the standard library to replicate whatever lodash is doing.
- [ ] [Moment](https://www.npmjs.com/package/moment) - It's not actually deprecated but much like lodash the community moved away from using it awhile ago. [You-Dont-Need-Momentjs](https://github.com/you-dont-need/You-Dont-Need-Momentjs) makes the case for removing it better than I can. I would argue though, this is a candidate for looking what we can do with the standard library before pulling in a replacement dependency
- [ ] pWaterfall -- It's not deprecated but we can easily rewrite the two places it's used in standard JS so we don't need this.
- [ ] [hapi/good](https://github.com/hapijs/good) Used to log ops statistics and request responses. Was deprecated in 2020 with no further support.

**Comment by Jozzey** [comment]
Repos that need sorting out:
- [ ] water-abstraction-ui
- [ ] water-abstraction-admin-ui
- [ ] water-abstraction-service
- [ ] water-abstraction-tactical-crm
- [ ] water-abstraction-import
- [ ] water-abstraction-tactical-idm
- [ ] water-abstraction-returns
- [ ] water-abstraction-permit-repository
- [ ] water-abstraction-reporting
- [ ] water-abstraction-helpers
- [ ] hapi-pg-rest-api

**Comment by Jozzey** [comment]
Still a bit of work to do in the water-abstraction-ui but the remaining work to be done has been added to a new issue https://github.com/DEFRA/water-abstraction-team/issues/78

**Comment by Cruikshanks** [comment]
When the current dev team first joined Water Abstraction we started creating issues for all the problems we saw. Whilst we were still getting to grips with the service it gave us opportunities to fix things and get stuck into the legacy code.

After a while, we became more confident and were able to get stuck into the critical feature at that time; SROC Supplementary billing. It also proved we could implement new and existing functionality in our [water-abstraction-system repo](https://github.com/DEFRA/water-abstraction-system) and integrate it with the existing apps. This, and the ongoing performance and stability issues with the legacy code have meant the whole delivery team are now onboard with us, over time, migrating _all_ functionality across.

So, as a dev team, we're instead going to focus our efforts on migrating functionality rather than fixing the legacy code. We'll continue to keep an eye on the packages the legacy code is using. But any that require effort to update will be ignored.
